textx/scoping/providers.py

```diff
@@ -201,20 +218,20 @@ def _find_referenced_obj(p, name, cls):
             Returns:
                 None or the found object
             """
-            ret = _find_obj_fqn(p, name, cls)
+            ret = _find_obj_fqn(p, name, cls, current_obj)
             if ret:
                 return ret
             while hasattr(p, "parent"):
                 p = p.parent
-                ret = _find_obj_fqn(p, name, cls)
+                ret = _find_obj_fqn(p, name, cls, current_obj)
                 if ret:
                     return ret
             return None
```

Comment:
I've just noticed this. In Python if function don't return anything it is implied that it returns `None`.

Reply:
OK. Nice to know. I removed two "return None" but also added one comment at the end of the while loop indicating that an implicit "return None" is expected in some cases.